### PR TITLE
Cancel query based on context

### PR DESCRIPTION
**Pull request by hitarth**
In QueryContext() user passes a context variable. When context is Done which could be due to multiple reasons like context deadline exceeded or user cancels the context deliberately, the query continues to run on presto server. This patch will send DELETE request to presto cluster to cancel the execution on cluster as soon as context is Done. This is useful in saving presto resources by killing the query which user wanted to cancel via presto-go-client. I have added unit test to test this as well. This is mentioned in issue #58

### CODE REVIEW

**Comment by BlueStalker** [comment]
Looks good to me. @fiorix Please also take a look at this, this will make the presto-go-client would be able to cancel the query, and it is already in our production environment. Also, we are using presto-go-client to develop our proxy layer, we have a few patches coming in the next and hope to collaborate with you guys!

**Comment by aweisberg** [comment]
@fiorix can you review/merge this?

**Comment by aweisberg** [comment]
Also @skogtwin and @winckler? Are you available to review/merge this?

File: presto/presto.go [hunk 1]
```diff
@@ -603,6 +603,19 @@ func (st *driverStmt) QueryContext(ctx context.Context, args []driver.NamedValue
 		stmt:    st,
 		nextURI: sr.NextURI,
 	}
+	completedChannel := make(chan struct{})
+	defer close(completedChannel)
+	go func() {
+		select {
+		case <-ctx.Done():
+			err := rows.Close()
```
Comment (skogtwin) [inline presto/presto.go:611]: The name of the `Close()` method is very confusing, it should've been `Cancel` or something along the lines. Not related to this PR tho...

**Pull request merged**